### ISSUE TITLE
ban an ip previously taken out of ban list after service rebooted

### ISSUE THREAD

**Issue by jandrich** [fail2ban/fail2ban#2615]
### Environment:

fail2ban 0.10.4-1 (EPEL stable repository)
fail2ban-hostsdeny
Red Hat 7.7
SSH with tcpwrappers (/etc/hosts.deny) NO firewall

### The issue:
fail2ban ban an ip previously taken out of ban list after service rebooted

#### Steps to reproduce
unban ip: "fail2ban-client set sshd unbainp 192.168.122.1"
I can connect again from 192.168.122.1
restart service: "systemctl restart fail2ban"
192.168.122.1 appears banned again

#### Any customizations done to /etc/fail2ban/ configuration
I add only file jail.local to /etc/fail2ban
```ini
[DEFAULT]
bantime = -1
ignoreip = 127.0.0.1

[sshd]
enabled = true
port = ssh
filter = sshd
logpath = /var/log/secure
maxretry = 3
findtime = 600
banaction = hostsdeny
```
#### Relevant parts of /var/log/fail2ban.log file:
```
2020-01-23 23:29:50,520 fail2ban.server         [1555]: INFO    Starting Fail2ban v0.10.4
2020-01-23 23:29:50,524 fail2ban.database       [1555]: INFO    Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
2020-01-23 23:29:50,525 fail2ban.jail           [1555]: INFO    Creating new jail 'sshd'
2020-01-23 23:29:50,539 fail2ban.jail           [1555]: INFO    Jail 'sshd' uses systemd {}
2020-01-23 23:29:50,539 fail2ban.jail           [1555]: INFO    Initiated 'systemd' backend
2020-01-23 23:29:50,540 fail2ban.filter         [1555]: INFO      maxLines: 1
2020-01-23 23:29:50,563 fail2ban.filtersystemd  [1555]: INFO    [sshd] Added journal match for: '_SYSTEMD_UNIT=sshd.service + _COMM=sshd'
2020-01-23 23:29:50,563 fail2ban.filter         [1555]: INFO      maxRetry: 3
2020-01-23 23:29:50,564 fail2ban.filter         [1555]: INFO      encoding: UTF-8
2020-01-23 23:29:50,564 fail2ban.filter         [1555]: INFO      findtime: 600
2020-01-23 23:29:50,564 fail2ban.actions        [1555]: INFO      banTime: 31536000
2020-01-23 23:29:50,566 fail2ban.jail           [1555]: INFO    Jail 'sshd' started
2020-01-23 23:30:16,344 fail2ban.filter         [1555]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:16
2020-01-23 23:30:18,945 fail2ban.filter         [1555]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
2020-01-23 23:30:18,946 fail2ban.filter         [1555]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
2020-01-23 23:30:18,946 fail2ban.filter         [1555]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
```
**Above. SSH connection attempts**
```
2020-01-23 23:30:19,222 fail2ban.actions        [1555]: NOTICE  [sshd] **Ban 192.168.122.1**
2020-01-23 23:30:20,882 fail2ban.filter         [1555]: INFO    [sshd] **Found 192.168.122.1** - 2020-01-23 23:30:20
2020-01-23 23:30:32,046 fail2ban.actions        [1555]: NOTICE  [sshd] **Unban 192.168.122.1**
2020-01-23 23:30:34,687 fail2ban.server         [1555]: INFO    Shutdown in progress...
2020-01-23 23:30:34,688 fail2ban.server         [1555]: INFO    Stopping all jails
2020-01-23 23:30:35,263 fail2ban.jail           [1555]: INFO    Jail 'sshd' stopped
2020-01-23 23:30:35,264 fail2ban.database       [1555]: INFO    Connection to database closed.
2020-01-23 23:30:35,265 fail2ban.server         [1555]: INFO    Exiting Fail2ban
2020-01-23 23:30:35,347 fail2ban.server         [1607]: INFO    --------------------------------------------------
2020-01-23 23:30:35,347 fail2ban.server         [1607]: INFO    Starting Fail2ban v0.10.4
2020-01-23 23:30:35,353 fail2ban.database       [1607]: INFO    Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
2020-01-23 23:30:35,354 fail2ban.jail           [1607]: INFO    Creating new jail 'sshd'
2020-01-23 23:30:35,368 fail2ban.jail           [1607]: INFO    Jail 'sshd' uses systemd {}
2020-01-23 23:30:35,368 fail2ban.jail           [1607]: INFO    Initiated 'systemd' backend
2020-01-23 23:30:35,369 fail2ban.filter         [1607]: INFO      maxLines: 1
2020-01-23 23:30:35,390 fail2ban.filtersystemd  [1607]: INFO    [sshd] Added journal match for: '_SYSTEMD_UNIT=sshd.service + _COMM=sshd'
2020-01-23 23:30:35,390 fail2ban.filter         [1607]: INFO      maxRetry: 3
2020-01-23 23:30:35,391 fail2ban.filter         [1607]: INFO      encoding: UTF-8
2020-01-23 23:30:35,391 fail2ban.filter         [1607]: INFO      findtime: 600
2020-01-23 23:30:35,391 fail2ban.actions        [1607]: INFO      banTime: 31536000
2020-01-23 23:30:35,395 fail2ban.jail           [1607]: INFO    Jail 'sshd' started
```
**Below. I didn't try any attempts
It's as if fail2ban matches were found in /var/log/secure, but it's not like that, because if I empty the file /var/log/secure it keeps happening**
```
2020-01-23 23:30:35,399 fail2ban.filter         [1607]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:16
2020-01-23 23:30:35,399 fail2ban.filter         [1607]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
2020-01-23 23:30:35,400 fail2ban.filter         [1607]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
2020-01-23 23:30:35,400 fail2ban.filter         [1607]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:18
2020-01-23 23:30:35,400 fail2ban.filter         [1607]: INFO    [sshd] Found 192.168.122.1 - 2020-01-23 23:30:20
2020-01-23 23:30:35,596 fail2ban.actions        [1607]: NOTICE  [sshd] **Ban 192.168.122.1**
```

**Comment by sebres** [comment]
> it's as if fail2ban matches were found in /var/log/secure, but it's not like that, because if I empty the file /var/log/secure it keeps happening

I don't think so because your backend used by jail is `systemd` (see `Jail 'sshd' uses systemd`), so fail2ban reads messages from systemd journal (it has basically nothing with file `/var/log/secure`).

If you had some file backend configured, you would not need to empty log-files also, because fail2ban stores last log position into database and would start monitoring from (so continues from this known position).

Just you seem to have systemd, and with this backend after restart fail2ban simply reads messages from `now - findtime` (back in journal).
Unfortunately it is impossible to disable this behavior at the moment (in opposite to all file-related backends, where it could be also disabled using `tail` option).

I can take a look whether some start position (or known time) can be retrieved from database here (so seekToTime of  this backend would jump to last known position) so it'd be able to ignore already processed messages.

**Comment by sebres** [comment]
fixed in 12b3ac684a637e2adfef05997d4aa3ad36b68970 for >= 0.10
